lib/hotel.rb

```diff
@@ -0,0 +1,162 @@
+
+require_relative "room"
+require_relative "reservation"
+require_relative "hotelblock"
+
+module HotelGroup
+  class Hotel
+    attr_accessor :id, :rooms, :reservations, :blocks
+
+    def initialize(directory: "./csv")
+      @id = 1
+      @rooms = Room.load_all(directory: directory)
+      @reservations = Reservation.load_all(directory: directory)
+      @blocks = HotelBlock.load_all(directory: directory)
+
+      connect_reservations
+      connect_blocks
+    end
+
+    def connect_blocks
+      blocks.each do |block|
+        room_array = []
+        block.rooms.each do |room_id|
+          room_obj = find_room(room_id)
+          room_array << room_obj
+          room_obj.add_block_id(block.id)
+          room_obj.block_price = room_obj.price - room_obj.price * block.discount
+        end
+        block.rooms = room_array
+      end
+      return blocks
+    end
+
+    def connect_reservations
+      reservations.each do |res|
+        room = find_room(res.room)
+        res.connect(room)
+        room.connect(res)
+      end
+
+      return reservations
+    end
+
+    def find_reservation(id)
+      Reservation.validate_id(id)
+      return @reservations.find { |res| res.id == id }
+    end
+
+    def find_room(id)
+      Room.validate_id(id)
+      return @rooms.find { |room| room.number == id }
+    end
+
+    def list_rooms
+      rooms.each do |room|
+        puts room.print_nicely
+      end
+    end
+
+    def list_blocks
+      blocks.each do |block|
+        puts block.print_nicely
+      end
+    end
+
+    def create_res_id
+      return reservations.count + 1
+    end
+
+    def create_block_id
+      return blocks.count + 1
+    end
+
+    def make_reservation(start_time, end_time, room: nil, block: nil)
+      if (start_time <=> end_time) == 1
+        raise ArgumentError, "End time must be later than start time"
```

Comment:
Let the reservation handle these errors, the hotel itself can worry about how to access info, the reservation can know about what makes a valid reservation.